costmap_2d/src/footprint.cpp

```diff
@@ -209,7 +209,7 @@ bool makeFootprintFromString(const std::string& footprint_string, std::vector<ge
 
 
 
-std::vector<geometry_msgs::Point> makeFootprintFromParams(ros::NodeHandle& nh)
+std::pair<std::vector<geometry_msgs::Point>, bool> makeFootprintFromParams(ros::NodeHandle& nh)
```

Comment:
S: Prefer `std::optional` for this usecase

Reply:
right, we make a .deb for this pkg, so returning a pair would have forced my to path all existing releases

changed, but using pointer